src/writer.rs

```diff
@@ -55,33 +54,31 @@ pub fn write<W: Write>(gpx: &Gpx, writer: W) -> Result<()> {
     Ok(())
 }
 
-fn write_xml_event<'a, W, E>(event: E, writer: &mut EventWriter<W>) -> Result<()>
+fn write_xml_event<'a, W, E>(event: E, writer: &mut EventWriter<W>) -> Result<(), GpxError>
 where
     W: Write,
     E: Into<XmlEvent<'a>>,
 {
-    writer
-        .write(event)
-        .chain_err(|| Error::from("error while writing gpx event"))
+    Ok(writer.write(event)?)
 }
 
-fn version_to_version_string(version: GpxVersion) -> Result<&'static str> {
+fn version_to_version_string(version: GpxVersion) -> Result<&'static str, GpxError> {
     match version {
         GpxVersion::Gpx10 => Ok("1.0"),
         GpxVersion::Gpx11 => Ok("1.1"),
-        version => Err(Error::from(format!("Unknown version {:?}", version))),
+        version => Err(GpxError::UnknownVersionError(version)),
```

Comment:
Getting the unknown version number in the error text was kind of nice, but not a big deal.

Reply:
Yeah the old way is nicer, but hopefully the enum makes it obvious what's gone wrong…

Reply:
Have also added a public doc string for the enum…